### PR TITLE
guide for migrating devices from facilities to metros

### LINKED ISSUE #83: Guide: Migrating Facility machines to metros
Users may want to update their existing facility based device deployments to use metro fields instead of facility fields.

Document the process as a guide, wether that involves simply changing the facility/metro fields and running `terraform apply`, or wether it involves `terraform plan; terraform detaint; terraform import; terraform apply` or some other combination.

What are the risks that should be avoided?

### PR DESCRIPTION
**Pull request by t0mk**
fixes #83 

Signed-off-by: Tomas Karasek <tom.to.the.k@gmail.com>

### CODE REVIEW

File: docs/guides/migration_guide_facilities_to_metros_devices.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+page_title: "Migrating devices from facilities to metros"
+description: |-
+  Migrating your device resources from facilities to metros
+---
+
+# Metros vs. Facilities
+
+In April 2021, Equinix Metal rolled out a new location concept - [metros](https://feedback.equinixmetal.com/changelog/new-metros-feature-live). A metro is an Equinix-wide concept for data centers which are grouped together geographically. Data centers within a metro share capacity and networking features. You can read more about metros at https://metal.equinix.com/developers/docs/locations/metros/.
+
+Until the metros introduction, resource deployment location used to be controlled by "facilty" - single data center with location code like "sv15" or "ny5". Metros group the facilities, and e.g. metro "sv" contains facility "sv15". If you specify a metro when creating a resource, it will (sometimes?) be deployed to one of the facilities in the metro group. You can then (sometimes?) find the deployed facility from a read-only attribute of the resource.
```
Comment (displague) [inline docs/guides/migration_guide_facilities_to_metros_devices.md:11]: no sometimes - both are always

File: docs/guides/migration_guide_facilities_to_metros_devices.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+## Changing your Terraform templates to use metros instead of facilities
+
+To take advantage of some of the features of the metro, you might want to change the configuration of your Terraform templates so that the devices have `metro` specified instead of `facility`. As both of the `metro` and `facility` are ForceNew paramters (a change will trigger re-creation of the resource), you should be cautious if you don't want to have the device destroyed. We updated the `metal_device` resource so that the change should be seamless, but please proceed with care.
+
+If you only want to change the `metal_device` resource specfication from a facility to a metro containing the facility, e.g. from facility "sv15" to metro "sv", it's enough to remove the `facilities` attribute, and add the `metro` attribute.
```
Comment (displague) [inline docs/guides/migration_guide_facilities_to_metros_devices.md:18]: > it's enough to remove the `facilities` attribute, and add the `metro` attribute

Is that true 😅 ?

File: docs/guides/migration_guide_facilities_to_metros_devices.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+To take advantage of some of the features of the metro, you might want to change the configuration of your Terraform templates so that the devices have `metro` specified instead of `facility`. As both of the `metro` and `facility` are ForceNew paramters (a change will trigger re-creation of the resource), you should be cautious if you don't want to have the device destroyed. We updated the `metal_device` resource so that the change should be seamless, but please proceed with care.
+
+If you only want to change the `metal_device` resource specfication from a facility to a metro containing the facility, e.g. from facility "sv15" to metro "sv", it's enough to remove the `facilities` attribute, and add the `metro` attribute. 
+
+The `metal_device` resource has the facilities input attributes.
+
+Given following configuration of a device deployed in the `sv15` facility.
+
+```hcl-terraform
```
Comment (displague) [inline docs/guides/migration_guide_facilities_to_metros_devices.md:24]: interesting - I've always used `hcl` and it worked.

File: docs/guides/migration_guide_facilities_to_metros_devices.md (hunk 2 above)
Comment (t0mk) [reply docs/guides/migration_guide_facilities_to_metros_devices.md]: Yes, I tested it with latest provider, and also I addressed it in a PR:
https://github.com/equinix/terraform-provider-metal/pull/46#issuecomment-810970346

File: docs/guides/migration_guide_facilities_to_metros_devices.md (hunk 3 above)
Comment (t0mk) [reply docs/guides/migration_guide_facilities_to_metros_devices.md:26]: I found it in linode or scaleway docs. It renders OK.

**Pull request merged**